Lib/typing.py

```diff
@@ -1675,13 +1675,66 @@ class _TypingEllipsis:
 
 _SPECIAL_NAMES = frozenset({
     '__abstractmethods__', '__annotations__', '__dict__', '__doc__',
-    '__init__', '__module__', '__new__', '__slots__',
-    '__subclasshook__', '__weakref__', '__class_getitem__',
-    '__match_args__',
+    '__module__', '__slots__', '__match_args__', '__qualname__',
+})
+_SPECIAL_CALLABLE_NAMES = frozenset({
+    '__init__', '__new__', '__subclasshook__','__class_getitem__', '__weakref__',
 })
 
 # These special attributes will be not collected as protocol members.
 EXCLUDED_ATTRIBUTES = _TYPING_INTERNALS | _SPECIAL_NAMES | {'_MutableMapping__marker'}
+EXCLUDED_MEMBERS = EXCLUDED_ATTRIBUTES | _SPECIAL_CALLABLE_NAMES
+
+
+def _get_local_members(namespace):
+    """Collect the specified attributes from a classes' namespace."""
+    annotations = namespace.get("__annotations__", {})
+    attrs = (set(namespace) | set(annotations)) - EXCLUDED_MEMBERS
```

Comment:
`dict_keys` instances are set-like and support set operations such as `|`. It's more efficient if you avoid converting the dictionary to a set.

```suggestion
    attrs = (namespace.keys() | annotations.keys()) - EXCLUDED_MEMBERS
```